File: WinPaletter/Classes/ExplorerPatcher.vb

Imports System.Net.Security

Public Class ExplorerPatcher
    Public Shared IsInstalled As Boolean = False
    Public UseStart10 As Boolean = False
    Public UseTaskbar10 As Boolean = False
    Public TaskbarButton10 As Boolean = False
    Public StartStyle As StartStyles

    Enum StartStyles
        NotRounded
        RoundedCornersFloatingMenu
        RoundedCornersDockedMenu
    End Enum

    Sub New()

        Try
            If My.Computer.Registry.CurrentUser.OpenSubKey("Software\ExplorerPatcher") IsNot Nothing Then
                IsInstalled = True
            Else
                IsInstalled = False
            End If
        Finally
            My.Computer.Registry.CurrentUser.Close()
        End Try

        If Not My.Settings.EP_Enabled_Force Then

            If IsInstalled And My.W11 Then

                Try
                    With My.Computer.Registry.CurrentUser.OpenSubKey("Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced")
                        UseStart10 = .GetValue("Start_ShowClassicMode")
                    End With
                Finally
                    My.Computer.Registry.CurrentUser.Close()
                End Try

                Try
                    With My.Computer.Registry.CurrentUser.OpenSubKey("Software\ExplorerPatcher")
                        UseTaskbar10 = .GetValue("OldTaskbar")
                        TaskbarButton10 = .GetValue("OrbStyle") = 0
                        StartStyle = .GetValue("StartUI_EnableRoundedCorners")
                    End With
                Finally
                    My.Computer.Registry.CurrentUser.Close()
                End Try
            Else
                UseStart10 = False
                UseTaskbar10 = False
                TaskbarButton10 = False
                StartStyle = StartStyles.NotRounded
            End If

        Else
            UseStart10 = My.Settings.EP_UseStart10
            UseTaskbar10 = My.Settings.EP_UseTaskbar10
            TaskbarButton10 = My.Settings.EP_TaskbarButton10
            StartStyle = My.Settings.EP_StartStyle
        End If


    End Sub

    Public Shared Function IsAllowed() As Boolean
        Dim condition0 As Boolean = My.W11 AndAlso My.Settings.EP_Enabled AndAlso IsInstalled
        Dim condition1 As Boolean = My.Settings.EP_Enabled_Force

        Return condition0 OrElse condition1
    End Function
End Class

File: WinPaletter/Classes/ExplorerPatcher.cs
using System;

namespace WinPaletter
{
    public class ExplorerPatcher
    {
        public static bool IsInstalled = false;
        public bool UseStart10 = false;
        public bool UseTaskbar10 = false;
        public bool TaskbarButton10 = false;
        public StartStyles StartStyle;

        public enum StartStyles
        {
            NotRounded,
            RoundedCornersFloatingMenu,
            RoundedCornersDockedMenu
        }

        public ExplorerPatcher()
        {

            try
            {
                if (Program.Computer.Registry.CurrentUser.OpenSubKey(@"Software\ExplorerPatcher") is not null)
                {
                    IsInstalled = true;
                }
                else
                {
                    IsInstalled = false;
                }
            }
            catch
            { }
            finally
            {
                Program.Computer.Registry.CurrentUser.Close();
            }

            if (!Program.Settings.ExplorerPatcher.Enabled_Force)
            {

                if (IsInstalled & Program.W11)
                {
                    UseStart10 = Convert.ToBoolean(Reg_IO.GetReg(@"HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced", "Start_ShowClassicMode", 0));
                    try
                    {
                        {
                            var temp = Program.Computer.Registry.CurrentUser.OpenSubKey(@"Software\ExplorerPatcher");
                            UseTaskbar10 = Convert.ToBoolean(temp.GetValue("OldTaskbar", true));
                            TaskbarButton10 = (int)temp.GetValue("OrbStyle", 0) == 0;
                            StartStyle = (StartStyles)Convert.ToInt32(temp.GetValue("StartUI_EnableRoundedCorners", StartStyles.NotRounded));
                        }
                    }
                    finally
                    {
                        Program.Computer.Registry.CurrentUser.Close();
                    }
                }
                else
                {
                    UseStart10 = false;
                    UseTaskbar10 = false;
                    TaskbarButton10 = false;
                    StartStyle = StartStyles.NotRounded;
                }
            }

            else
            {
                UseStart10 = Program.Settings.ExplorerPatcher.UseStart10;
                UseTaskbar10 = Program.Settings.ExplorerPatcher.UseTaskbar10;
                TaskbarButton10 = Program.Settings.ExplorerPatcher.TaskbarButton10;
                StartStyle = Program.Settings.ExplorerPatcher.StartStyle;
            }


        }

        public static bool IsAllowed()
        {
            bool condition0 = Program.W11 && Program.Settings.ExplorerPatcher.Enabled && IsInstalled;
            bool condition1 = Program.Settings.ExplorerPatcher.Enabled_Force;

            return condition0 || condition1;
        }
    }
}